### ISSUE TITLE
Introduce a `regex` node

### ISSUE THREAD

**Issue by dglazkov** [google/labs-prototypes#70]
JSONAta is cool, but for many cases, a simple regular expression might work better.

**Comment by proales** [comment]
What is the thinking here? A node that runs a regex and passes on if it matches or some other kind of node like a node that asks the LLM for a regex and applies it to the text somehow?

**Comment by dglazkov** [comment]
yeah, in the simplest form it probably takes in two inputs: value and expression, and returns matches:true/false if matches or doesn't match.

Or maybe a replace form is more useful.

@kex103 you played with this some in your prototype. WDYT?
